// @pulumi/gcp/securitycenter/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SourceIamPolicy = exports.SourceIamMember = exports.SourceIamBinding = exports.Source = exports.NotificationConfig = exports.MuteConfig = exports.InstanceIamPolicy = exports.InstanceIamMember = exports.InstanceIamBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.InstanceIamBinding = null;
utilities.lazyLoad(exports, ["InstanceIamBinding"], () => require("./instanceIamBinding"));
exports.InstanceIamMember = null;
utilities.lazyLoad(exports, ["InstanceIamMember"], () => require("./instanceIamMember"));
exports.InstanceIamPolicy = null;
utilities.lazyLoad(exports, ["InstanceIamPolicy"], () => require("./instanceIamPolicy"));
exports.MuteConfig = null;
utilities.lazyLoad(exports, ["MuteConfig"], () => require("./muteConfig"));
exports.NotificationConfig = null;
utilities.lazyLoad(exports, ["NotificationConfig"], () => require("./notificationConfig"));
exports.Source = null;
utilities.lazyLoad(exports, ["Source"], () => require("./source"));
exports.SourceIamBinding = null;
utilities.lazyLoad(exports, ["SourceIamBinding"], () => require("./sourceIamBinding"));
exports.SourceIamMember = null;
utilities.lazyLoad(exports, ["SourceIamMember"], () => require("./sourceIamMember"));
exports.SourceIamPolicy = null;
utilities.lazyLoad(exports, ["SourceIamPolicy"], () => require("./sourceIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:securitycenter/instanceIamBinding:InstanceIamBinding":
                return new exports.InstanceIamBinding(name, undefined, { urn });
            case "gcp:securitycenter/instanceIamMember:InstanceIamMember":
                return new exports.InstanceIamMember(name, undefined, { urn });
            case "gcp:securitycenter/instanceIamPolicy:InstanceIamPolicy":
                return new exports.InstanceIamPolicy(name, undefined, { urn });
            case "gcp:securitycenter/muteConfig:MuteConfig":
                return new exports.MuteConfig(name, undefined, { urn });
            case "gcp:securitycenter/notificationConfig:NotificationConfig":
                return new exports.NotificationConfig(name, undefined, { urn });
            case "gcp:securitycenter/source:Source":
                return new exports.Source(name, undefined, { urn });
            case "gcp:securitycenter/sourceIamBinding:SourceIamBinding":
                return new exports.SourceIamBinding(name, undefined, { urn });
            case "gcp:securitycenter/sourceIamMember:SourceIamMember":
                return new exports.SourceIamMember(name, undefined, { urn });
            case "gcp:securitycenter/sourceIamPolicy:SourceIamPolicy":
                return new exports.SourceIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "securitycenter/instanceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/instanceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/instanceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/muteConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/notificationConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/source", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/sourceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/sourceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "securitycenter/sourceIamPolicy", _module);
