// @pulumi/gcp/networkservices/endpointPolicy.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EndpointPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 * ### Network Services Endpoint Policy Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.EndpointPolicy("default", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     type: "SIDECAR_PROXY",
 *     trafficPortSelector: {
 *         ports: ["8081"],
 *     },
 *     endpointMatcher: {
 *         metadataLabelMatcher: {
 *             metadataLabelMatchCriteria: "MATCH_ANY",
 *             metadataLabels: [{
 *                 labelName: "foo",
 *                 labelValue: "bar",
 *             }],
 *         },
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Network Services Endpoint Policy Empty Match
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.EndpointPolicy("default", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     type: "SIDECAR_PROXY",
 *     trafficPortSelector: {
 *         ports: ["8081"],
 *     },
 *     endpointMatcher: {
 *         metadataLabelMatcher: {
 *             metadataLabelMatchCriteria: "MATCH_ANY",
 *         },
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * EndpointPolicy can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/endpointPolicy:EndpointPolicy default projects/{{project}}/locations/global/endpointPolicies/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/endpointPolicy:EndpointPolicy default {{project}}/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/endpointPolicy:EndpointPolicy default {{name}}
 * ```
 */
class EndpointPolicy extends pulumi.CustomResource {
    /**
     * Get an existing EndpointPolicy resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EndpointPolicy(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of EndpointPolicy.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EndpointPolicy.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["authorizationPolicy"] = state ? state.authorizationPolicy : undefined;
            resourceInputs["clientTlsPolicy"] = state ? state.clientTlsPolicy : undefined;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["endpointMatcher"] = state ? state.endpointMatcher : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["serverTlsPolicy"] = state ? state.serverTlsPolicy : undefined;
            resourceInputs["trafficPortSelector"] = state ? state.trafficPortSelector : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.endpointMatcher === undefined) && !opts.urn) {
                throw new Error("Missing required property 'endpointMatcher'");
            }
            if ((!args || args.type === undefined) && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["authorizationPolicy"] = args ? args.authorizationPolicy : undefined;
            resourceInputs["clientTlsPolicy"] = args ? args.clientTlsPolicy : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["endpointMatcher"] = args ? args.endpointMatcher : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["serverTlsPolicy"] = args ? args.serverTlsPolicy : undefined;
            resourceInputs["trafficPortSelector"] = args ? args.trafficPortSelector : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EndpointPolicy.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EndpointPolicy = EndpointPolicy;
/** @internal */
EndpointPolicy.__pulumiType = 'gcp:networkservices/endpointPolicy:EndpointPolicy';
