// @pulumi/gcp/networkservices/httpRoute.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HttpRoute = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 * ### Network Services Http Route Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         matches: [{
 *             queryParameters: [{
 *                 queryParameter: "key",
 *                 exactMatch: "value",
 *             }],
 *             fullPathMatch: "example",
 *         }],
 *     }],
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Network Services Http Route Matches And Actions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         matches: [
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     regexMatch: "header-value",
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     exactMatch: "value",
 *                 }],
 *                 prefixMatch: "example",
 *                 ignoreCase: false,
 *             },
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     presentMatch: true,
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     regexMatch: "value",
 *                 }],
 *                 regexMatch: "example",
 *                 ignoreCase: false,
 *             },
 *             {
 *                 headers: [{
 *                     header: "header",
 *                     invertMatch: false,
 *                     presentMatch: true,
 *                 }],
 *                 queryParameters: [{
 *                     queryParameter: "key",
 *                     presentMatch: true,
 *                 }],
 *                 fullPathMatch: "example",
 *                 ignoreCase: false,
 *             },
 *         ],
 *         action: {
 *             redirect: {
 *                 hostRedirect: "new-host",
 *                 pathRedirect: "new-path",
 *                 prefixRewrite: "new-prefix",
 *                 httpsRedirect: true,
 *                 stripQuery: true,
 *                 portRedirect: 8081,
 *             },
 *             urlRewrite: {
 *                 pathPrefixRewrite: "new-prefix",
 *                 hostRewrite: "new-host",
 *             },
 *             retryPolicy: {
 *                 retryConditions: ["server_error"],
 *                 numRetries: 1,
 *                 perTryTimeout: "1s",
 *             },
 *             requestMirrorPolicy: {
 *                 destination: {
 *                     serviceName: "new",
 *                     weight: 1,
 *                 },
 *             },
 *             corsPolicy: {
 *                 allowOrigins: ["example"],
 *                 allowMethods: [
 *                     "GET",
 *                     "PUT",
 *                 ],
 *                 allowHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 exposeHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 maxAge: "1s",
 *                 allowCredentials: true,
 *                 disabled: false,
 *             },
 *         },
 *     }],
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Network Services Http Route Actions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const _default = new gcp.networkservices.HttpRoute("default", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     rules: [{
 *         action: {
 *             faultInjectionPolicy: {
 *                 delay: {
 *                     fixedDelay: "1s",
 *                     percentage: 1,
 *                 },
 *                 abort: {
 *                     httpStatus: 500,
 *                     percentage: 1,
 *                 },
 *             },
 *             urlRewrite: {
 *                 pathPrefixRewrite: "new-prefix",
 *                 hostRewrite: "new-host",
 *             },
 *             retryPolicy: {
 *                 retryConditions: ["server_error"],
 *                 numRetries: 1,
 *                 perTryTimeout: "1s",
 *             },
 *             requestMirrorPolicy: {
 *                 destination: {
 *                     serviceName: "new",
 *                     weight: 1,
 *                 },
 *             },
 *             corsPolicy: {
 *                 allowOrigins: ["example"],
 *                 allowMethods: [
 *                     "GET",
 *                     "PUT",
 *                 ],
 *                 allowHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 exposeHeaders: [
 *                     "version",
 *                     "type",
 *                 ],
 *                 maxAge: "1s",
 *                 allowCredentials: true,
 *                 disabled: false,
 *             },
 *             requestHeaderModifier: {
 *                 set: {
 *                     version: "1",
 *                     type: "json",
 *                 },
 *                 add: {
 *                     "minor-version": "1",
 *                 },
 *                 removes: ["arg"],
 *             },
 *             responseHeaderModifier: {
 *                 set: {
 *                     version: "1",
 *                     type: "json",
 *                 },
 *                 add: {
 *                     "minor-version": "1",
 *                 },
 *                 removes: ["removearg"],
 *             },
 *         },
 *     }],
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Network Services Http Route Mesh Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultMesh = new gcp.networkservices.Mesh("defaultMesh", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHttpRoute = new gcp.networkservices.HttpRoute("defaultHttpRoute", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "my description",
 *     hostnames: ["example"],
 *     meshes: [defaultMesh.id],
 *     rules: [{
 *         matches: [{
 *             queryParameters: [{
 *                 queryParameter: "key",
 *                 exactMatch: "value",
 *             }],
 *             fullPathMatch: "example",
 *         }],
 *     }],
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * HttpRoute can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/httpRoute:HttpRoute default projects/{{project}}/locations/global/httpRoutes/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/httpRoute:HttpRoute default {{project}}/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:networkservices/httpRoute:HttpRoute default {{name}}
 * ```
 */
class HttpRoute extends pulumi.CustomResource {
    /**
     * Get an existing HttpRoute resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HttpRoute(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HttpRoute.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HttpRoute.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["gateways"] = state ? state.gateways : undefined;
            resourceInputs["hostnames"] = state ? state.hostnames : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["meshes"] = state ? state.meshes : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["rules"] = state ? state.rules : undefined;
            resourceInputs["selfLink"] = state ? state.selfLink : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.hostnames === undefined) && !opts.urn) {
                throw new Error("Missing required property 'hostnames'");
            }
            if ((!args || args.rules === undefined) && !opts.urn) {
                throw new Error("Missing required property 'rules'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["gateways"] = args ? args.gateways : undefined;
            resourceInputs["hostnames"] = args ? args.hostnames : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["meshes"] = args ? args.meshes : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["rules"] = args ? args.rules : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["selfLink"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HttpRoute.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HttpRoute = HttpRoute;
/** @internal */
HttpRoute.__pulumiType = 'gcp:networkservices/httpRoute:HttpRoute';
