// @pulumi/gcp/iot/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RegistryIamPolicy = exports.RegistryIamMember = exports.RegistryIamBinding = exports.Registry = exports.Device = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Device = null;
utilities.lazyLoad(exports, ["Device"], () => require("./device"));
exports.Registry = null;
utilities.lazyLoad(exports, ["Registry"], () => require("./registry"));
exports.RegistryIamBinding = null;
utilities.lazyLoad(exports, ["RegistryIamBinding"], () => require("./registryIamBinding"));
exports.RegistryIamMember = null;
utilities.lazyLoad(exports, ["RegistryIamMember"], () => require("./registryIamMember"));
exports.RegistryIamPolicy = null;
utilities.lazyLoad(exports, ["RegistryIamPolicy"], () => require("./registryIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:iot/device:Device":
                return new exports.Device(name, undefined, { urn });
            case "gcp:iot/registry:Registry":
                return new exports.Registry(name, undefined, { urn });
            case "gcp:iot/registryIamBinding:RegistryIamBinding":
                return new exports.RegistryIamBinding(name, undefined, { urn });
            case "gcp:iot/registryIamMember:RegistryIamMember":
                return new exports.RegistryIamMember(name, undefined, { urn });
            case "gcp:iot/registryIamPolicy:RegistryIamPolicy":
                return new exports.RegistryIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "iot/device", _module);
pulumi.runtime.registerResourceModule("gcp", "iot/registry", _module);
pulumi.runtime.registerResourceModule("gcp", "iot/registryIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "iot/registryIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "iot/registryIamPolicy", _module);
