// @pulumi/gcp/gkeonprem/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.VMwareNodePool = exports.VMwareCluster = exports.BareMetalNodePool = exports.BareMetalCluster = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BareMetalCluster = null;
utilities.lazyLoad(exports, ["BareMetalCluster"], () => require("./bareMetalCluster"));
exports.BareMetalNodePool = null;
utilities.lazyLoad(exports, ["BareMetalNodePool"], () => require("./bareMetalNodePool"));
exports.VMwareCluster = null;
utilities.lazyLoad(exports, ["VMwareCluster"], () => require("./vmwareCluster"));
exports.VMwareNodePool = null;
utilities.lazyLoad(exports, ["VMwareNodePool"], () => require("./vmwareNodePool"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:gkeonprem/bareMetalCluster:BareMetalCluster":
                return new exports.BareMetalCluster(name, undefined, { urn });
            case "gcp:gkeonprem/bareMetalNodePool:BareMetalNodePool":
                return new exports.BareMetalNodePool(name, undefined, { urn });
            case "gcp:gkeonprem/vMwareCluster:VMwareCluster":
                return new exports.VMwareCluster(name, undefined, { urn });
            case "gcp:gkeonprem/vMwareNodePool:VMwareNodePool":
                return new exports.VMwareNodePool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "gkeonprem/bareMetalCluster", _module);
pulumi.runtime.registerResourceModule("gcp", "gkeonprem/bareMetalNodePool", _module);
pulumi.runtime.registerResourceModule("gcp", "gkeonprem/vMwareCluster", _module);
pulumi.runtime.registerResourceModule("gcp", "gkeonprem/vMwareNodePool", _module);
