// @pulumi/gcp/healthcare/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Hl7StoreIamPolicy = exports.Hl7StoreIamMember = exports.Hl7StoreIamBinding = exports.Hl7Store = exports.FhirStoreIamPolicy = exports.FhirStoreIamMember = exports.FhirStoreIamBinding = exports.FhirStore = exports.DicomStoreIamPolicy = exports.DicomStoreIamMember = exports.DicomStoreIamBinding = exports.DicomStore = exports.DatasetIamPolicy = exports.DatasetIamMember = exports.DatasetIamBinding = exports.Dataset = exports.ConsentStoreIamPolicy = exports.ConsentStoreIamMember = exports.ConsentStoreIamBinding = exports.ConsentStore = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.ConsentStore = null;
utilities.lazyLoad(exports, ["ConsentStore"], () => require("./consentStore"));
exports.ConsentStoreIamBinding = null;
utilities.lazyLoad(exports, ["ConsentStoreIamBinding"], () => require("./consentStoreIamBinding"));
exports.ConsentStoreIamMember = null;
utilities.lazyLoad(exports, ["ConsentStoreIamMember"], () => require("./consentStoreIamMember"));
exports.ConsentStoreIamPolicy = null;
utilities.lazyLoad(exports, ["ConsentStoreIamPolicy"], () => require("./consentStoreIamPolicy"));
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.DatasetIamBinding = null;
utilities.lazyLoad(exports, ["DatasetIamBinding"], () => require("./datasetIamBinding"));
exports.DatasetIamMember = null;
utilities.lazyLoad(exports, ["DatasetIamMember"], () => require("./datasetIamMember"));
exports.DatasetIamPolicy = null;
utilities.lazyLoad(exports, ["DatasetIamPolicy"], () => require("./datasetIamPolicy"));
exports.DicomStore = null;
utilities.lazyLoad(exports, ["DicomStore"], () => require("./dicomStore"));
exports.DicomStoreIamBinding = null;
utilities.lazyLoad(exports, ["DicomStoreIamBinding"], () => require("./dicomStoreIamBinding"));
exports.DicomStoreIamMember = null;
utilities.lazyLoad(exports, ["DicomStoreIamMember"], () => require("./dicomStoreIamMember"));
exports.DicomStoreIamPolicy = null;
utilities.lazyLoad(exports, ["DicomStoreIamPolicy"], () => require("./dicomStoreIamPolicy"));
exports.FhirStore = null;
utilities.lazyLoad(exports, ["FhirStore"], () => require("./fhirStore"));
exports.FhirStoreIamBinding = null;
utilities.lazyLoad(exports, ["FhirStoreIamBinding"], () => require("./fhirStoreIamBinding"));
exports.FhirStoreIamMember = null;
utilities.lazyLoad(exports, ["FhirStoreIamMember"], () => require("./fhirStoreIamMember"));
exports.FhirStoreIamPolicy = null;
utilities.lazyLoad(exports, ["FhirStoreIamPolicy"], () => require("./fhirStoreIamPolicy"));
exports.Hl7Store = null;
utilities.lazyLoad(exports, ["Hl7Store"], () => require("./hl7Store"));
exports.Hl7StoreIamBinding = null;
utilities.lazyLoad(exports, ["Hl7StoreIamBinding"], () => require("./hl7StoreIamBinding"));
exports.Hl7StoreIamMember = null;
utilities.lazyLoad(exports, ["Hl7StoreIamMember"], () => require("./hl7StoreIamMember"));
exports.Hl7StoreIamPolicy = null;
utilities.lazyLoad(exports, ["Hl7StoreIamPolicy"], () => require("./hl7StoreIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:healthcare/consentStore:ConsentStore":
                return new exports.ConsentStore(name, undefined, { urn });
            case "gcp:healthcare/consentStoreIamBinding:ConsentStoreIamBinding":
                return new exports.ConsentStoreIamBinding(name, undefined, { urn });
            case "gcp:healthcare/consentStoreIamMember:ConsentStoreIamMember":
                return new exports.ConsentStoreIamMember(name, undefined, { urn });
            case "gcp:healthcare/consentStoreIamPolicy:ConsentStoreIamPolicy":
                return new exports.ConsentStoreIamPolicy(name, undefined, { urn });
            case "gcp:healthcare/dataset:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "gcp:healthcare/datasetIamBinding:DatasetIamBinding":
                return new exports.DatasetIamBinding(name, undefined, { urn });
            case "gcp:healthcare/datasetIamMember:DatasetIamMember":
                return new exports.DatasetIamMember(name, undefined, { urn });
            case "gcp:healthcare/datasetIamPolicy:DatasetIamPolicy":
                return new exports.DatasetIamPolicy(name, undefined, { urn });
            case "gcp:healthcare/dicomStore:DicomStore":
                return new exports.DicomStore(name, undefined, { urn });
            case "gcp:healthcare/dicomStoreIamBinding:DicomStoreIamBinding":
                return new exports.DicomStoreIamBinding(name, undefined, { urn });
            case "gcp:healthcare/dicomStoreIamMember:DicomStoreIamMember":
                return new exports.DicomStoreIamMember(name, undefined, { urn });
            case "gcp:healthcare/dicomStoreIamPolicy:DicomStoreIamPolicy":
                return new exports.DicomStoreIamPolicy(name, undefined, { urn });
            case "gcp:healthcare/fhirStore:FhirStore":
                return new exports.FhirStore(name, undefined, { urn });
            case "gcp:healthcare/fhirStoreIamBinding:FhirStoreIamBinding":
                return new exports.FhirStoreIamBinding(name, undefined, { urn });
            case "gcp:healthcare/fhirStoreIamMember:FhirStoreIamMember":
                return new exports.FhirStoreIamMember(name, undefined, { urn });
            case "gcp:healthcare/fhirStoreIamPolicy:FhirStoreIamPolicy":
                return new exports.FhirStoreIamPolicy(name, undefined, { urn });
            case "gcp:healthcare/hl7Store:Hl7Store":
                return new exports.Hl7Store(name, undefined, { urn });
            case "gcp:healthcare/hl7StoreIamBinding:Hl7StoreIamBinding":
                return new exports.Hl7StoreIamBinding(name, undefined, { urn });
            case "gcp:healthcare/hl7StoreIamMember:Hl7StoreIamMember":
                return new exports.Hl7StoreIamMember(name, undefined, { urn });
            case "gcp:healthcare/hl7StoreIamPolicy:Hl7StoreIamPolicy":
                return new exports.Hl7StoreIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "healthcare/consentStore", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/consentStoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/consentStoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/consentStoreIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/dataset", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/datasetIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/datasetIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/datasetIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/dicomStore", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/dicomStoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/dicomStoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/dicomStoreIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/fhirStore", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/fhirStoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/fhirStoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/fhirStoreIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/hl7Store", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/hl7StoreIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/hl7StoreIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "healthcare/hl7StoreIamPolicy", _module);
