// @pulumi/gcp/accessapproval/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getProjectServiceAccountOutput = exports.getProjectServiceAccount = exports.getOrganizationServiceAccountOutput = exports.getOrganizationServiceAccount = exports.getFolderServiceAccountOutput = exports.getFolderServiceAccount = void 0;
const utilities = require("../utilities");
exports.getFolderServiceAccount = null;
exports.getFolderServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getFolderServiceAccount", "getFolderServiceAccountOutput"], () => require("./getFolderServiceAccount"));
exports.getOrganizationServiceAccount = null;
exports.getOrganizationServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getOrganizationServiceAccount", "getOrganizationServiceAccountOutput"], () => require("./getOrganizationServiceAccount"));
exports.getProjectServiceAccount = null;
exports.getProjectServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getProjectServiceAccount", "getProjectServiceAccountOutput"], () => require("./getProjectServiceAccount"));
