// @pulumi/gcp/iam/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkloadIdentityPoolProvider = exports.WorkloadIdentityPool = exports.WorkforcePoolProvider = exports.WorkforcePool = exports.getWorkloadIdentityPoolProviderOutput = exports.getWorkloadIdentityPoolProvider = exports.getWorkloadIdentityPoolOutput = exports.getWorkloadIdentityPool = exports.getTestablePermissionsOutput = exports.getTestablePermissions = exports.getRuleOutput = exports.getRule = exports.DenyPolicy = exports.AccessBoundaryPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccessBoundaryPolicy = null;
utilities.lazyLoad(exports, ["AccessBoundaryPolicy"], () => require("./accessBoundaryPolicy"));
exports.DenyPolicy = null;
utilities.lazyLoad(exports, ["DenyPolicy"], () => require("./denyPolicy"));
exports.getRule = null;
exports.getRuleOutput = null;
utilities.lazyLoad(exports, ["getRule", "getRuleOutput"], () => require("./getRule"));
exports.getTestablePermissions = null;
exports.getTestablePermissionsOutput = null;
utilities.lazyLoad(exports, ["getTestablePermissions", "getTestablePermissionsOutput"], () => require("./getTestablePermissions"));
exports.getWorkloadIdentityPool = null;
exports.getWorkloadIdentityPoolOutput = null;
utilities.lazyLoad(exports, ["getWorkloadIdentityPool", "getWorkloadIdentityPoolOutput"], () => require("./getWorkloadIdentityPool"));
exports.getWorkloadIdentityPoolProvider = null;
exports.getWorkloadIdentityPoolProviderOutput = null;
utilities.lazyLoad(exports, ["getWorkloadIdentityPoolProvider", "getWorkloadIdentityPoolProviderOutput"], () => require("./getWorkloadIdentityPoolProvider"));
exports.WorkforcePool = null;
utilities.lazyLoad(exports, ["WorkforcePool"], () => require("./workforcePool"));
exports.WorkforcePoolProvider = null;
utilities.lazyLoad(exports, ["WorkforcePoolProvider"], () => require("./workforcePoolProvider"));
exports.WorkloadIdentityPool = null;
utilities.lazyLoad(exports, ["WorkloadIdentityPool"], () => require("./workloadIdentityPool"));
exports.WorkloadIdentityPoolProvider = null;
utilities.lazyLoad(exports, ["WorkloadIdentityPoolProvider"], () => require("./workloadIdentityPoolProvider"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:iam/accessBoundaryPolicy:AccessBoundaryPolicy":
                return new exports.AccessBoundaryPolicy(name, undefined, { urn });
            case "gcp:iam/denyPolicy:DenyPolicy":
                return new exports.DenyPolicy(name, undefined, { urn });
            case "gcp:iam/workforcePool:WorkforcePool":
                return new exports.WorkforcePool(name, undefined, { urn });
            case "gcp:iam/workforcePoolProvider:WorkforcePoolProvider":
                return new exports.WorkforcePoolProvider(name, undefined, { urn });
            case "gcp:iam/workloadIdentityPool:WorkloadIdentityPool":
                return new exports.WorkloadIdentityPool(name, undefined, { urn });
            case "gcp:iam/workloadIdentityPoolProvider:WorkloadIdentityPoolProvider":
                return new exports.WorkloadIdentityPoolProvider(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "iam/accessBoundaryPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iam/denyPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "iam/workforcePool", _module);
pulumi.runtime.registerResourceModule("gcp", "iam/workforcePoolProvider", _module);
pulumi.runtime.registerResourceModule("gcp", "iam/workloadIdentityPool", _module);
pulumi.runtime.registerResourceModule("gcp", "iam/workloadIdentityPoolProvider", _module);
