// @pulumi/gcp/cloudfunctions/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getFunctionOutput = exports.getFunction = exports.FunctionIamPolicy = exports.FunctionIamMember = exports.FunctionIamBinding = exports.Function = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Function = null;
utilities.lazyLoad(exports, ["Function"], () => require("./function"));
exports.FunctionIamBinding = null;
utilities.lazyLoad(exports, ["FunctionIamBinding"], () => require("./functionIamBinding"));
exports.FunctionIamMember = null;
utilities.lazyLoad(exports, ["FunctionIamMember"], () => require("./functionIamMember"));
exports.FunctionIamPolicy = null;
utilities.lazyLoad(exports, ["FunctionIamPolicy"], () => require("./functionIamPolicy"));
exports.getFunction = null;
exports.getFunctionOutput = null;
utilities.lazyLoad(exports, ["getFunction", "getFunctionOutput"], () => require("./getFunction"));
__exportStar(require("./zMixins"), exports);
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudfunctions/function:Function":
                return new exports.Function(name, undefined, { urn });
            case "gcp:cloudfunctions/functionIamBinding:FunctionIamBinding":
                return new exports.FunctionIamBinding(name, undefined, { urn });
            case "gcp:cloudfunctions/functionIamMember:FunctionIamMember":
                return new exports.FunctionIamMember(name, undefined, { urn });
            case "gcp:cloudfunctions/functionIamPolicy:FunctionIamPolicy":
                return new exports.FunctionIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudfunctions/function", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudfunctions/functionIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudfunctions/functionIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudfunctions/functionIamPolicy", _module);
