// @pulumi/gcp/beyondcorp/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getAppGatewayOutput = exports.getAppGateway = exports.getAppConnectorOutput = exports.getAppConnector = exports.getAppConnectionOutput = exports.getAppConnection = exports.AppGateway = exports.AppConnector = exports.AppConnection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AppConnection = null;
utilities.lazyLoad(exports, ["AppConnection"], () => require("./appConnection"));
exports.AppConnector = null;
utilities.lazyLoad(exports, ["AppConnector"], () => require("./appConnector"));
exports.AppGateway = null;
utilities.lazyLoad(exports, ["AppGateway"], () => require("./appGateway"));
exports.getAppConnection = null;
exports.getAppConnectionOutput = null;
utilities.lazyLoad(exports, ["getAppConnection", "getAppConnectionOutput"], () => require("./getAppConnection"));
exports.getAppConnector = null;
exports.getAppConnectorOutput = null;
utilities.lazyLoad(exports, ["getAppConnector", "getAppConnectorOutput"], () => require("./getAppConnector"));
exports.getAppGateway = null;
exports.getAppGatewayOutput = null;
utilities.lazyLoad(exports, ["getAppGateway", "getAppGatewayOutput"], () => require("./getAppGateway"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:beyondcorp/appConnection:AppConnection":
                return new exports.AppConnection(name, undefined, { urn });
            case "gcp:beyondcorp/appConnector:AppConnector":
                return new exports.AppConnector(name, undefined, { urn });
            case "gcp:beyondcorp/appGateway:AppGateway":
                return new exports.AppGateway(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appConnection", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appConnector", _module);
pulumi.runtime.registerResourceModule("gcp", "beyondcorp/appGateway", _module);
