// @pulumi/gcp/serviceaccount/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Key = exports.IAMPolicy = exports.IAMMember = exports.IAMBinding = exports.getAccountKeyOutput = exports.getAccountKey = exports.getAccountJwtOutput = exports.getAccountJwt = exports.getAccountIdTokenOutput = exports.getAccountIdToken = exports.getAccountAccessTokenOutput = exports.getAccountAccessToken = exports.getAccountOutput = exports.getAccount = exports.Account = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Account = null;
utilities.lazyLoad(exports, ["Account"], () => require("./account"));
exports.getAccount = null;
exports.getAccountOutput = null;
utilities.lazyLoad(exports, ["getAccount", "getAccountOutput"], () => require("./getAccount"));
exports.getAccountAccessToken = null;
exports.getAccountAccessTokenOutput = null;
utilities.lazyLoad(exports, ["getAccountAccessToken", "getAccountAccessTokenOutput"], () => require("./getAccountAccessToken"));
exports.getAccountIdToken = null;
exports.getAccountIdTokenOutput = null;
utilities.lazyLoad(exports, ["getAccountIdToken", "getAccountIdTokenOutput"], () => require("./getAccountIdToken"));
exports.getAccountJwt = null;
exports.getAccountJwtOutput = null;
utilities.lazyLoad(exports, ["getAccountJwt", "getAccountJwtOutput"], () => require("./getAccountJwt"));
exports.getAccountKey = null;
exports.getAccountKeyOutput = null;
utilities.lazyLoad(exports, ["getAccountKey", "getAccountKeyOutput"], () => require("./getAccountKey"));
exports.IAMBinding = null;
utilities.lazyLoad(exports, ["IAMBinding"], () => require("./iambinding"));
exports.IAMMember = null;
utilities.lazyLoad(exports, ["IAMMember"], () => require("./iammember"));
exports.IAMPolicy = null;
utilities.lazyLoad(exports, ["IAMPolicy"], () => require("./iampolicy"));
exports.Key = null;
utilities.lazyLoad(exports, ["Key"], () => require("./key"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:serviceAccount/account:Account":
                return new exports.Account(name, undefined, { urn });
            case "gcp:serviceAccount/iAMBinding:IAMBinding":
                return new exports.IAMBinding(name, undefined, { urn });
            case "gcp:serviceAccount/iAMMember:IAMMember":
                return new exports.IAMMember(name, undefined, { urn });
            case "gcp:serviceAccount/iAMPolicy:IAMPolicy":
                return new exports.IAMPolicy(name, undefined, { urn });
            case "gcp:serviceAccount/key:Key":
                return new exports.Key(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "serviceAccount/account", _module);
pulumi.runtime.registerResourceModule("gcp", "serviceAccount/iAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "serviceAccount/iAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "serviceAccount/iAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "serviceAccount/key", _module);
