// @pulumi/gcp/cloudbuildv2/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Repository = exports.ConnectionIAMPolicy = exports.ConnectionIAMMember = exports.ConnectionIAMBinding = exports.Connection = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionIAMBinding = null;
utilities.lazyLoad(exports, ["ConnectionIAMBinding"], () => require("./connectionIAMBinding"));
exports.ConnectionIAMMember = null;
utilities.lazyLoad(exports, ["ConnectionIAMMember"], () => require("./connectionIAMMember"));
exports.ConnectionIAMPolicy = null;
utilities.lazyLoad(exports, ["ConnectionIAMPolicy"], () => require("./connectionIAMPolicy"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudbuildv2/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMBinding:ConnectionIAMBinding":
                return new exports.ConnectionIAMBinding(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMMember:ConnectionIAMMember":
                return new exports.ConnectionIAMMember(name, undefined, { urn });
            case "gcp:cloudbuildv2/connectionIAMPolicy:ConnectionIAMPolicy":
                return new exports.ConnectionIAMPolicy(name, undefined, { urn });
            case "gcp:cloudbuildv2/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connection", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/connectionIAMPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuildv2/repository", _module);
