// @pulumi/gcp/firestore/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Index = exports.Field = exports.Document = exports.Database = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Database = null;
utilities.lazyLoad(exports, ["Database"], () => require("./database"));
exports.Document = null;
utilities.lazyLoad(exports, ["Document"], () => require("./document"));
exports.Field = null;
utilities.lazyLoad(exports, ["Field"], () => require("./field"));
exports.Index = null;
utilities.lazyLoad(exports, ["Index"], () => require("./index_"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:firestore/database:Database":
                return new exports.Database(name, undefined, { urn });
            case "gcp:firestore/document:Document":
                return new exports.Document(name, undefined, { urn });
            case "gcp:firestore/field:Field":
                return new exports.Field(name, undefined, { urn });
            case "gcp:firestore/index:Index":
                return new exports.Index(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "firestore/database", _module);
pulumi.runtime.registerResourceModule("gcp", "firestore/document", _module);
pulumi.runtime.registerResourceModule("gcp", "firestore/field", _module);
pulumi.runtime.registerResourceModule("gcp", "firestore/index", _module);
