// @pulumi/gcp/bigqueryanalyticshub/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListingIamPolicy = exports.ListingIamMember = exports.ListingIamBinding = exports.Listing = exports.DataExchangeIamPolicy = exports.DataExchangeIamMember = exports.DataExchangeIamBinding = exports.DataExchange = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataExchange = null;
utilities.lazyLoad(exports, ["DataExchange"], () => require("./dataExchange"));
exports.DataExchangeIamBinding = null;
utilities.lazyLoad(exports, ["DataExchangeIamBinding"], () => require("./dataExchangeIamBinding"));
exports.DataExchangeIamMember = null;
utilities.lazyLoad(exports, ["DataExchangeIamMember"], () => require("./dataExchangeIamMember"));
exports.DataExchangeIamPolicy = null;
utilities.lazyLoad(exports, ["DataExchangeIamPolicy"], () => require("./dataExchangeIamPolicy"));
exports.Listing = null;
utilities.lazyLoad(exports, ["Listing"], () => require("./listing"));
exports.ListingIamBinding = null;
utilities.lazyLoad(exports, ["ListingIamBinding"], () => require("./listingIamBinding"));
exports.ListingIamMember = null;
utilities.lazyLoad(exports, ["ListingIamMember"], () => require("./listingIamMember"));
exports.ListingIamPolicy = null;
utilities.lazyLoad(exports, ["ListingIamPolicy"], () => require("./listingIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigqueryanalyticshub/dataExchange:DataExchange":
                return new exports.DataExchange(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamBinding:DataExchangeIamBinding":
                return new exports.DataExchangeIamBinding(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamMember:DataExchangeIamMember":
                return new exports.DataExchangeIamMember(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/dataExchangeIamPolicy:DataExchangeIamPolicy":
                return new exports.DataExchangeIamPolicy(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listing:Listing":
                return new exports.Listing(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamBinding:ListingIamBinding":
                return new exports.ListingIamBinding(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamMember:ListingIamMember":
                return new exports.ListingIamMember(name, undefined, { urn });
            case "gcp:bigqueryanalyticshub/listingIamPolicy:ListingIamPolicy":
                return new exports.ListingIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchange", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/dataExchangeIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listing", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigqueryanalyticshub/listingIamPolicy", _module);
