// @pulumi/gcp/cloudtasks/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueIamPolicy = exports.QueueIamMember = exports.QueueIamBinding = exports.Queue = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Queue = null;
utilities.lazyLoad(exports, ["Queue"], () => require("./queue"));
exports.QueueIamBinding = null;
utilities.lazyLoad(exports, ["QueueIamBinding"], () => require("./queueIamBinding"));
exports.QueueIamMember = null;
utilities.lazyLoad(exports, ["QueueIamMember"], () => require("./queueIamMember"));
exports.QueueIamPolicy = null;
utilities.lazyLoad(exports, ["QueueIamPolicy"], () => require("./queueIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudtasks/queue:Queue":
                return new exports.Queue(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamBinding:QueueIamBinding":
                return new exports.QueueIamBinding(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamMember:QueueIamMember":
                return new exports.QueueIamMember(name, undefined, { urn });
            case "gcp:cloudtasks/queueIamPolicy:QueueIamPolicy":
                return new exports.QueueIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queue", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudtasks/queueIamPolicy", _module);
