// @pulumi/gcp/endpoints/service.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * This resource creates and rolls out a Cloud Endpoints service using OpenAPI or gRPC.  View the relevant docs for [OpenAPI](https://cloud.google.com/endpoints/docs/openapi/) and [gRPC](https://cloud.google.com/endpoints/docs/grpc/).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as fs from "fs";
 * import * as gcp from "@pulumi/gcp";
 *
 * const openapiService = new gcp.endpoints.Service("openapiService", {
 *     serviceName: "api-name.endpoints.project-id.cloud.goog",
 *     project: "project-id",
 *     openapiConfig: fs.readFileSync("openapi_spec.yml"),
 * });
 * const grpcService = new gcp.endpoints.Service("grpcService", {
 *     serviceName: "api-name.endpoints.project-id.cloud.goog",
 *     project: "project-id",
 *     grpcConfig: fs.readFileSync("service_spec.yml"),
 *     protocOutputBase64: Buffer.from(fs.readFileSync("compiled_descriptor_file.pb"), 'binary').toString('base64'),
 * });
 * ```
 *
 * The example in `examples/endpoints_on_compute_engine` shows the API from the quickstart running on a Compute Engine VM and reachable through Cloud Endpoints, which may also be useful.
 *
 * ## Import
 *
 * This resource does not support import.
 */
class Service extends pulumi.CustomResource {
    /**
     * Get an existing Service resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Service(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Service.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Service.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["apis"] = state ? state.apis : undefined;
            resourceInputs["configId"] = state ? state.configId : undefined;
            resourceInputs["dnsAddress"] = state ? state.dnsAddress : undefined;
            resourceInputs["endpoints"] = state ? state.endpoints : undefined;
            resourceInputs["grpcConfig"] = state ? state.grpcConfig : undefined;
            resourceInputs["openapiConfig"] = state ? state.openapiConfig : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["protocOutputBase64"] = state ? state.protocOutputBase64 : undefined;
            resourceInputs["serviceName"] = state ? state.serviceName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.serviceName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'serviceName'");
            }
            resourceInputs["grpcConfig"] = args ? args.grpcConfig : undefined;
            resourceInputs["openapiConfig"] = args ? args.openapiConfig : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["protocOutputBase64"] = args ? args.protocOutputBase64 : undefined;
            resourceInputs["serviceName"] = args ? args.serviceName : undefined;
            resourceInputs["apis"] = undefined /*out*/;
            resourceInputs["configId"] = undefined /*out*/;
            resourceInputs["dnsAddress"] = undefined /*out*/;
            resourceInputs["endpoints"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Service.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Service = Service;
/** @internal */
Service.__pulumiType = 'gcp:endpoints/service:Service';
