// @pulumi/gcp/bigquerydatapolicy/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataPolicyIamPolicy = exports.DataPolicyIamMember = exports.DataPolicyIamBinding = exports.DataPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.DataPolicy = null;
utilities.lazyLoad(exports, ["DataPolicy"], () => require("./dataPolicy"));
exports.DataPolicyIamBinding = null;
utilities.lazyLoad(exports, ["DataPolicyIamBinding"], () => require("./dataPolicyIamBinding"));
exports.DataPolicyIamMember = null;
utilities.lazyLoad(exports, ["DataPolicyIamMember"], () => require("./dataPolicyIamMember"));
exports.DataPolicyIamPolicy = null;
utilities.lazyLoad(exports, ["DataPolicyIamPolicy"], () => require("./dataPolicyIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigquerydatapolicy/dataPolicy:DataPolicy":
                return new exports.DataPolicy(name, undefined, { urn });
            case "gcp:bigquerydatapolicy/dataPolicyIamBinding:DataPolicyIamBinding":
                return new exports.DataPolicyIamBinding(name, undefined, { urn });
            case "gcp:bigquerydatapolicy/dataPolicyIamMember:DataPolicyIamMember":
                return new exports.DataPolicyIamMember(name, undefined, { urn });
            case "gcp:bigquerydatapolicy/dataPolicyIamPolicy:DataPolicyIamPolicy":
                return new exports.DataPolicyIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigquerydatapolicy/dataPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquerydatapolicy/dataPolicyIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquerydatapolicy/dataPolicyIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquerydatapolicy/dataPolicyIamPolicy", _module);
