// @pulumi/gcp/cloudasset/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectFeed = exports.OrganizationFeed = exports.getResourcesSearchAllOutput = exports.getResourcesSearchAll = exports.FolderFeed = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.FolderFeed = null;
utilities.lazyLoad(exports, ["FolderFeed"], () => require("./folderFeed"));
exports.getResourcesSearchAll = null;
exports.getResourcesSearchAllOutput = null;
utilities.lazyLoad(exports, ["getResourcesSearchAll", "getResourcesSearchAllOutput"], () => require("./getResourcesSearchAll"));
exports.OrganizationFeed = null;
utilities.lazyLoad(exports, ["OrganizationFeed"], () => require("./organizationFeed"));
exports.ProjectFeed = null;
utilities.lazyLoad(exports, ["ProjectFeed"], () => require("./projectFeed"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudasset/folderFeed:FolderFeed":
                return new exports.FolderFeed(name, undefined, { urn });
            case "gcp:cloudasset/organizationFeed:OrganizationFeed":
                return new exports.OrganizationFeed(name, undefined, { urn });
            case "gcp:cloudasset/projectFeed:ProjectFeed":
                return new exports.ProjectFeed(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudasset/folderFeed", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudasset/organizationFeed", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudasset/projectFeed", _module);
