// @pulumi/gcp/vertex/aiFeatureStoreEntityType.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AiFeatureStoreEntityType = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * An entity type is a type of object in a system that needs to be modeled and have stored information about. For example, driver is an entity type, and driver0 is an instance of an entity type driver.
 *
 * To get more information about FeaturestoreEntitytype, see:
 *
 * * [API documentation](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.featurestores.entityTypes)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/vertex-ai/docs)
 *
 * ## Example Usage
 * ### Vertex Ai Featurestore Entitytype
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const featurestore = new gcp.vertex.AiFeatureStore("featurestore", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     region: "us-central1",
 *     onlineServingConfig: {
 *         fixedNodeCount: 2,
 *     },
 *     encryptionSpec: {
 *         kmsKeyName: "kms-name",
 *     },
 * });
 * const entity = new gcp.vertex.AiFeatureStoreEntityType("entity", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     description: "test description",
 *     featurestore: featurestore.id,
 *     monitoringConfig: {
 *         snapshotAnalysis: {
 *             disabled: false,
 *             monitoringIntervalDays: 1,
 *             stalenessDays: 21,
 *         },
 *         numericalThresholdConfig: {
 *             value: 0.8,
 *         },
 *         categoricalThresholdConfig: {
 *             value: 10,
 *         },
 *         importFeaturesAnalysis: {
 *             state: "ENABLED",
 *             anomalyDetectionBaseline: "PREVIOUS_IMPORT_FEATURES_STATS",
 *         },
 *     },
 * });
 * ```
 * ### Vertex Ai Featurestore Entitytype With Beta Fields
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const featurestore = new gcp.vertex.AiFeatureStore("featurestore", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     region: "us-central1",
 *     onlineServingConfig: {
 *         fixedNodeCount: 2,
 *     },
 *     encryptionSpec: {
 *         kmsKeyName: "kms-name",
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const entity = new gcp.vertex.AiFeatureStoreEntityType("entity", {
 *     labels: {
 *         foo: "bar",
 *     },
 *     featurestore: featurestore.id,
 *     monitoringConfig: {
 *         snapshotAnalysis: {
 *             disabled: false,
 *             monitoringInterval: "86400s",
 *         },
 *         categoricalThresholdConfig: {
 *             value: 0.3,
 *         },
 *         numericalThresholdConfig: {
 *             value: 0.3,
 *         },
 *     },
 *     offlineStorageTtlDays: 30,
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * FeaturestoreEntitytype can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:vertex/aiFeatureStoreEntityType:AiFeatureStoreEntityType default {{featurestore}}/entityTypes/{{name}}
 * ```
 */
class AiFeatureStoreEntityType extends pulumi.CustomResource {
    /**
     * Get an existing AiFeatureStoreEntityType resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AiFeatureStoreEntityType(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AiFeatureStoreEntityType.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AiFeatureStoreEntityType.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createTime"] = state ? state.createTime : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["etag"] = state ? state.etag : undefined;
            resourceInputs["featurestore"] = state ? state.featurestore : undefined;
            resourceInputs["labels"] = state ? state.labels : undefined;
            resourceInputs["monitoringConfig"] = state ? state.monitoringConfig : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["offlineStorageTtlDays"] = state ? state.offlineStorageTtlDays : undefined;
            resourceInputs["region"] = state ? state.region : undefined;
            resourceInputs["updateTime"] = state ? state.updateTime : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.featurestore === undefined) && !opts.urn) {
                throw new Error("Missing required property 'featurestore'");
            }
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["featurestore"] = args ? args.featurestore : undefined;
            resourceInputs["labels"] = args ? args.labels : undefined;
            resourceInputs["monitoringConfig"] = args ? args.monitoringConfig : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["offlineStorageTtlDays"] = args ? args.offlineStorageTtlDays : undefined;
            resourceInputs["createTime"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["region"] = undefined /*out*/;
            resourceInputs["updateTime"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AiFeatureStoreEntityType.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AiFeatureStoreEntityType = AiFeatureStoreEntityType;
/** @internal */
AiFeatureStoreEntityType.__pulumiType = 'gcp:vertex/aiFeatureStoreEntityType:AiFeatureStoreEntityType';
