// @pulumi/gcp/folder/accessApprovalSettings.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessApprovalSettings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Access Approval enables you to require your explicit approval whenever Google support and engineering need to access your customer content.
 *
 * To get more information about FolderSettings, see:
 *
 * * [API documentation](https://cloud.google.com/access-approval/docs/reference/rest/v1/folders)
 *
 * ## Example Usage
 * ### Folder Access Approval Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const myFolder = new gcp.organizations.Folder("myFolder", {
 *     displayName: "my-folder",
 *     parent: "organizations/123456789",
 * });
 * const folderAccessApproval = new gcp.folder.AccessApprovalSettings("folderAccessApproval", {
 *     folderId: myFolder.folderId,
 *     notificationEmails: [
 *         "testuser@example.com",
 *         "example.user@example.com",
 *     ],
 *     enrolledServices: [{
 *         cloudProduct: "all",
 *     }],
 * });
 * ```
 * ### Folder Access Approval Active Key Version
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const myFolder = new gcp.organizations.Folder("myFolder", {
 *     displayName: "my-folder",
 *     parent: "organizations/123456789",
 * });
 * const myProject = new gcp.organizations.Project("myProject", {
 *     projectId: "your-project-id",
 *     folderId: myFolder.name,
 * });
 * const keyRing = new gcp.kms.KeyRing("keyRing", {
 *     location: "global",
 *     project: myProject.projectId,
 * });
 * const cryptoKey = new gcp.kms.CryptoKey("cryptoKey", {
 *     keyRing: keyRing.id,
 *     purpose: "ASYMMETRIC_SIGN",
 *     versionTemplate: {
 *         algorithm: "EC_SIGN_P384_SHA384",
 *     },
 * });
 * const serviceAccount = gcp.accessapproval.getFolderServiceAccountOutput({
 *     folderId: myFolder.folderId,
 * });
 * const iam = new gcp.kms.CryptoKeyIAMMember("iam", {
 *     cryptoKeyId: cryptoKey.id,
 *     role: "roles/cloudkms.signerVerifier",
 *     member: serviceAccount.apply(serviceAccount => `serviceAccount:${serviceAccount.accountEmail}`),
 * });
 * const cryptoKeyVersion = gcp.kms.getKMSCryptoKeyVersionOutput({
 *     cryptoKey: cryptoKey.id,
 * });
 * const folderAccessApproval = new gcp.folder.AccessApprovalSettings("folderAccessApproval", {
 *     folderId: myFolder.folderId,
 *     activeKeyVersion: cryptoKeyVersion.apply(cryptoKeyVersion => cryptoKeyVersion.name),
 *     enrolledServices: [{
 *         cloudProduct: "all",
 *     }],
 * }, {
 *     dependsOn: [iam],
 * });
 * ```
 *
 * ## Import
 *
 * FolderSettings can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:folder/accessApprovalSettings:AccessApprovalSettings default folders/{{folder_id}}/accessApprovalSettings
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:folder/accessApprovalSettings:AccessApprovalSettings default {{folder_id}}
 * ```
 */
class AccessApprovalSettings extends pulumi.CustomResource {
    /**
     * Get an existing AccessApprovalSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccessApprovalSettings(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AccessApprovalSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccessApprovalSettings.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["activeKeyVersion"] = state ? state.activeKeyVersion : undefined;
            resourceInputs["ancestorHasActiveKeyVersion"] = state ? state.ancestorHasActiveKeyVersion : undefined;
            resourceInputs["enrolledAncestor"] = state ? state.enrolledAncestor : undefined;
            resourceInputs["enrolledServices"] = state ? state.enrolledServices : undefined;
            resourceInputs["folderId"] = state ? state.folderId : undefined;
            resourceInputs["invalidKeyVersion"] = state ? state.invalidKeyVersion : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["notificationEmails"] = state ? state.notificationEmails : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.enrolledServices === undefined) && !opts.urn) {
                throw new Error("Missing required property 'enrolledServices'");
            }
            if ((!args || args.folderId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'folderId'");
            }
            resourceInputs["activeKeyVersion"] = args ? args.activeKeyVersion : undefined;
            resourceInputs["enrolledServices"] = args ? args.enrolledServices : undefined;
            resourceInputs["folderId"] = args ? args.folderId : undefined;
            resourceInputs["notificationEmails"] = args ? args.notificationEmails : undefined;
            resourceInputs["ancestorHasActiveKeyVersion"] = undefined /*out*/;
            resourceInputs["enrolledAncestor"] = undefined /*out*/;
            resourceInputs["invalidKeyVersion"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccessApprovalSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccessApprovalSettings = AccessApprovalSettings;
/** @internal */
AccessApprovalSettings.__pulumiType = 'gcp:folder/accessApprovalSettings:AccessApprovalSettings';
