// @pulumi/gcp/composer/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.getImageVersionsOutput = exports.getImageVersions = exports.getEnvironmentOutput = exports.getEnvironment = exports.Environment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Environment = null;
utilities.lazyLoad(exports, ["Environment"], () => require("./environment"));
exports.getEnvironment = null;
exports.getEnvironmentOutput = null;
utilities.lazyLoad(exports, ["getEnvironment", "getEnvironmentOutput"], () => require("./getEnvironment"));
exports.getImageVersions = null;
exports.getImageVersionsOutput = null;
utilities.lazyLoad(exports, ["getImageVersions", "getImageVersionsOutput"], () => require("./getImageVersions"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:composer/environment:Environment":
                return new exports.Environment(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "composer/environment", _module);
