// @pulumi/gcp/billing/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubAccount = exports.Budget = exports.AccountIamPolicy = exports.AccountIamMember = exports.AccountIamBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AccountIamBinding = null;
utilities.lazyLoad(exports, ["AccountIamBinding"], () => require("./accountIamBinding"));
exports.AccountIamMember = null;
utilities.lazyLoad(exports, ["AccountIamMember"], () => require("./accountIamMember"));
exports.AccountIamPolicy = null;
utilities.lazyLoad(exports, ["AccountIamPolicy"], () => require("./accountIamPolicy"));
exports.Budget = null;
utilities.lazyLoad(exports, ["Budget"], () => require("./budget"));
exports.SubAccount = null;
utilities.lazyLoad(exports, ["SubAccount"], () => require("./subAccount"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:billing/accountIamBinding:AccountIamBinding":
                return new exports.AccountIamBinding(name, undefined, { urn });
            case "gcp:billing/accountIamMember:AccountIamMember":
                return new exports.AccountIamMember(name, undefined, { urn });
            case "gcp:billing/accountIamPolicy:AccountIamPolicy":
                return new exports.AccountIamPolicy(name, undefined, { urn });
            case "gcp:billing/budget:Budget":
                return new exports.Budget(name, undefined, { urn });
            case "gcp:billing/subAccount:SubAccount":
                return new exports.SubAccount(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/accountIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/budget", _module);
pulumi.runtime.registerResourceModule("gcp", "billing/subAccount", _module);
