// @pulumi/gcp/firebase/androidApp.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AndroidApp = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 * ### Firebase Android App Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const basic = new gcp.firebase.AndroidApp("basic", {
 *     project: "my-project-name",
 *     displayName: "Display Name Basic",
 *     packageName: "",
 *     sha1Hashes: ["2145bdf698b8715039bd0e83f2069bed435ac21c"],
 *     sha256Hashes: ["2145bdf698b8715039bd0e83f2069bed435ac21ca1b2c3d4e5f6123456789abc"],
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * AndroidApp can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:firebase/androidApp:AndroidApp default projects/{{project}}/androidApps/{{appId}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/androidApp:AndroidApp default {{project}}/{{appId}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/androidApp:AndroidApp default androidApps/{{appId}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/androidApp:AndroidApp default {{appId}}
 * ```
 */
class AndroidApp extends pulumi.CustomResource {
    /**
     * Get an existing AndroidApp resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AndroidApp(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of AndroidApp.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AndroidApp.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["appId"] = state ? state.appId : undefined;
            resourceInputs["deletionPolicy"] = state ? state.deletionPolicy : undefined;
            resourceInputs["displayName"] = state ? state.displayName : undefined;
            resourceInputs["etag"] = state ? state.etag : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["packageName"] = state ? state.packageName : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["sha1Hashes"] = state ? state.sha1Hashes : undefined;
            resourceInputs["sha256Hashes"] = state ? state.sha256Hashes : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.displayName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'displayName'");
            }
            resourceInputs["deletionPolicy"] = args ? args.deletionPolicy : undefined;
            resourceInputs["displayName"] = args ? args.displayName : undefined;
            resourceInputs["packageName"] = args ? args.packageName : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["sha1Hashes"] = args ? args.sha1Hashes : undefined;
            resourceInputs["sha256Hashes"] = args ? args.sha256Hashes : undefined;
            resourceInputs["appId"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AndroidApp.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AndroidApp = AndroidApp;
/** @internal */
AndroidApp.__pulumiType = 'gcp:firebase/androidApp:AndroidApp';
