// @pulumi/gcp/firebase/hostingRelease.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostingRelease = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 * ### Firebasehosting Release In Site
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-id",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingVersion = new gcp.firebase.HostingVersion("defaultHostingVersion", {
 *     siteId: defaultHostingSite.siteId,
 *     config: {
 *         redirects: [{
 *             glob: "/google/**",
 *             statusCode: 302,
 *             location: "https://www.google.com",
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     versionName: defaultHostingVersion.name,
 *     message: "Test release",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Firebasehosting Release In Channel
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-with-channel",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingVersion = new gcp.firebase.HostingVersion("defaultHostingVersion", {
 *     siteId: defaultHostingSite.siteId,
 *     config: {
 *         redirects: [{
 *             glob: "/google/**",
 *             statusCode: 302,
 *             location: "https://www.google.com",
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingChannel = new gcp.firebase.HostingChannel("defaultHostingChannel", {
 *     siteId: defaultHostingSite.siteId,
 *     channelId: "channel-id",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     channelId: defaultHostingChannel.channelId,
 *     versionName: defaultHostingVersion.name,
 *     message: "Test release in channel",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Firebasehosting Release Disable
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-id",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     type: "SITE_DISABLE",
 *     message: "Take down site",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * Release can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingRelease:HostingRelease default sites/{{site_id}}/channels/{{channel_id}}/releases/{{release_id}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingRelease:HostingRelease default sites/{{site_id}}/releases/{{release_id}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingRelease:HostingRelease default {{site_id}}/{{channel_id}}/{{release_id}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingRelease:HostingRelease default {{site_id}}/{{release_id}}
 * ```
 */
class HostingRelease extends pulumi.CustomResource {
    /**
     * Get an existing HostingRelease resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HostingRelease(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HostingRelease.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HostingRelease.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["channelId"] = state ? state.channelId : undefined;
            resourceInputs["message"] = state ? state.message : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["releaseId"] = state ? state.releaseId : undefined;
            resourceInputs["siteId"] = state ? state.siteId : undefined;
            resourceInputs["type"] = state ? state.type : undefined;
            resourceInputs["versionName"] = state ? state.versionName : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.siteId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'siteId'");
            }
            resourceInputs["channelId"] = args ? args.channelId : undefined;
            resourceInputs["message"] = args ? args.message : undefined;
            resourceInputs["siteId"] = args ? args.siteId : undefined;
            resourceInputs["type"] = args ? args.type : undefined;
            resourceInputs["versionName"] = args ? args.versionName : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["releaseId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HostingRelease.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HostingRelease = HostingRelease;
/** @internal */
HostingRelease.__pulumiType = 'gcp:firebase/hostingRelease:HostingRelease';
