// @pulumi/gcp/firebase/hostingVersion.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostingVersion = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * ## Example Usage
 * ### Firebasehosting Version Redirect
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-id",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingVersion = new gcp.firebase.HostingVersion("defaultHostingVersion", {
 *     siteId: defaultHostingSite.siteId,
 *     config: {
 *         redirects: [{
 *             glob: "/google/**",
 *             statusCode: 302,
 *             location: "https://www.google.com",
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     versionName: defaultHostingVersion.name,
 *     message: "Redirect to Google",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Firebasehosting Version Cloud Run
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-id",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultService = new gcp.cloudrunv2.Service("defaultService", {
 *     project: "my-project-name",
 *     location: "us-central1",
 *     ingress: "INGRESS_TRAFFIC_ALL",
 *     template: {
 *         containers: [{
 *             image: "us-docker.pkg.dev/cloudrun/container/hello",
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingVersion = new gcp.firebase.HostingVersion("defaultHostingVersion", {
 *     siteId: defaultHostingSite.siteId,
 *     config: {
 *         rewrites: [{
 *             glob: "/hello/**",
 *             run: {
 *                 serviceId: defaultService.name,
 *                 region: defaultService.location,
 *             },
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     versionName: defaultHostingVersion.name,
 *     message: "Cloud Run Integration",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 * ### Firebasehosting Version Cloud Functions
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const defaultHostingSite = new gcp.firebase.HostingSite("defaultHostingSite", {
 *     project: "my-project-name",
 *     siteId: "site-id",
 * }, {
 *     provider: google_beta,
 * });
 * const bucket = new gcp.storage.Bucket("bucket", {
 *     project: "my-project-name",
 *     location: "US",
 *     uniformBucketLevelAccess: true,
 * }, {
 *     provider: google_beta,
 * });
 * const object = new gcp.storage.BucketObject("object", {
 *     bucket: bucket.name,
 *     source: new pulumi.asset.FileAsset("function-source.zip"),
 * }, {
 *     provider: google_beta,
 * });
 * // Add path to the zipped function source code
 * const _function = new gcp.cloudfunctions.Function("function", {
 *     project: "my-project-name",
 *     description: "A Cloud Function connected to Firebase Hosing",
 *     runtime: "nodejs16",
 *     availableMemoryMb: 128,
 *     sourceArchiveBucket: bucket.name,
 *     sourceArchiveObject: object.name,
 *     triggerHttp: true,
 *     entryPoint: "helloHttp",
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingVersion = new gcp.firebase.HostingVersion("defaultHostingVersion", {
 *     siteId: defaultHostingSite.siteId,
 *     config: {
 *         rewrites: [{
 *             glob: "/hello/**",
 *             "function": _function.name,
 *         }],
 *     },
 * }, {
 *     provider: google_beta,
 * });
 * const defaultHostingRelease = new gcp.firebase.HostingRelease("defaultHostingRelease", {
 *     siteId: defaultHostingSite.siteId,
 *     versionName: defaultHostingVersion.name,
 *     message: "Cloud Functions Integration",
 * }, {
 *     provider: google_beta,
 * });
 * ```
 *
 * ## Import
 *
 * Version can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingVersion:HostingVersion default sites/{{site_id}}/versions/{{version_id}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:firebase/hostingVersion:HostingVersion default {{site_id}}/{{version_id}}
 * ```
 */
class HostingVersion extends pulumi.CustomResource {
    /**
     * Get an existing HostingVersion resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HostingVersion(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of HostingVersion.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HostingVersion.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["config"] = state ? state.config : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["siteId"] = state ? state.siteId : undefined;
            resourceInputs["versionId"] = state ? state.versionId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.siteId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'siteId'");
            }
            resourceInputs["config"] = args ? args.config : undefined;
            resourceInputs["siteId"] = args ? args.siteId : undefined;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["versionId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HostingVersion.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HostingVersion = HostingVersion;
/** @internal */
HostingVersion.__pulumiType = 'gcp:firebase/hostingVersion:HostingVersion';
