// @pulumi/gcp/cloudbuild/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkerPool = exports.Trigger = exports.getTriggerOutput = exports.getTrigger = exports.BitbucketServerConfig = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.BitbucketServerConfig = null;
utilities.lazyLoad(exports, ["BitbucketServerConfig"], () => require("./bitbucketServerConfig"));
exports.getTrigger = null;
exports.getTriggerOutput = null;
utilities.lazyLoad(exports, ["getTrigger", "getTriggerOutput"], () => require("./getTrigger"));
exports.Trigger = null;
utilities.lazyLoad(exports, ["Trigger"], () => require("./trigger"));
exports.WorkerPool = null;
utilities.lazyLoad(exports, ["WorkerPool"], () => require("./workerPool"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:cloudbuild/bitbucketServerConfig:BitbucketServerConfig":
                return new exports.BitbucketServerConfig(name, undefined, { urn });
            case "gcp:cloudbuild/trigger:Trigger":
                return new exports.Trigger(name, undefined, { urn });
            case "gcp:cloudbuild/workerPool:WorkerPool":
                return new exports.WorkerPool(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "cloudbuild/bitbucketServerConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuild/trigger", _module);
pulumi.runtime.registerResourceModule("gcp", "cloudbuild/workerPool", _module);
