// @pulumi/gcp/bigtable/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TableIamPolicy = exports.TableIamMember = exports.TableIamBinding = exports.Table = exports.InstanceIamPolicy = exports.InstanceIamMember = exports.InstanceIamBinding = exports.Instance = exports.GCPolicy = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.GCPolicy = null;
utilities.lazyLoad(exports, ["GCPolicy"], () => require("./gcpolicy"));
exports.Instance = null;
utilities.lazyLoad(exports, ["Instance"], () => require("./instance"));
exports.InstanceIamBinding = null;
utilities.lazyLoad(exports, ["InstanceIamBinding"], () => require("./instanceIamBinding"));
exports.InstanceIamMember = null;
utilities.lazyLoad(exports, ["InstanceIamMember"], () => require("./instanceIamMember"));
exports.InstanceIamPolicy = null;
utilities.lazyLoad(exports, ["InstanceIamPolicy"], () => require("./instanceIamPolicy"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
exports.TableIamBinding = null;
utilities.lazyLoad(exports, ["TableIamBinding"], () => require("./tableIamBinding"));
exports.TableIamMember = null;
utilities.lazyLoad(exports, ["TableIamMember"], () => require("./tableIamMember"));
exports.TableIamPolicy = null;
utilities.lazyLoad(exports, ["TableIamPolicy"], () => require("./tableIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigtable/gCPolicy:GCPolicy":
                return new exports.GCPolicy(name, undefined, { urn });
            case "gcp:bigtable/instance:Instance":
                return new exports.Instance(name, undefined, { urn });
            case "gcp:bigtable/instanceIamBinding:InstanceIamBinding":
                return new exports.InstanceIamBinding(name, undefined, { urn });
            case "gcp:bigtable/instanceIamMember:InstanceIamMember":
                return new exports.InstanceIamMember(name, undefined, { urn });
            case "gcp:bigtable/instanceIamPolicy:InstanceIamPolicy":
                return new exports.InstanceIamPolicy(name, undefined, { urn });
            case "gcp:bigtable/table:Table":
                return new exports.Table(name, undefined, { urn });
            case "gcp:bigtable/tableIamBinding:TableIamBinding":
                return new exports.TableIamBinding(name, undefined, { urn });
            case "gcp:bigtable/tableIamMember:TableIamMember":
                return new exports.TableIamMember(name, undefined, { urn });
            case "gcp:bigtable/tableIamPolicy:TableIamPolicy":
                return new exports.TableIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigtable/gCPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instance", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/instanceIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/table", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigtable/tableIamPolicy", _module);
