// @pulumi/gcp/accesscontextmanager/gcpUserAccessBinding.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GcpUserAccessBinding = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * Restricts access to Cloud Console and Google Cloud APIs for a set of users using Context-Aware Access.
 *
 * To get more information about GcpUserAccessBinding, see:
 *
 * * [API documentation](https://cloud.google.com/access-context-manager/docs/reference/rest/v1/organizations.gcpUserAccessBindings)
 *
 * ## Example Usage
 *
 * ## Import
 *
 * GcpUserAccessBinding can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:accesscontextmanager/gcpUserAccessBinding:GcpUserAccessBinding default {{name}}
 * ```
 */
class GcpUserAccessBinding extends pulumi.CustomResource {
    /**
     * Get an existing GcpUserAccessBinding resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GcpUserAccessBinding(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GcpUserAccessBinding.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GcpUserAccessBinding.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accessLevels"] = state ? state.accessLevels : undefined;
            resourceInputs["groupKey"] = state ? state.groupKey : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["organizationId"] = state ? state.organizationId : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.accessLevels === undefined) && !opts.urn) {
                throw new Error("Missing required property 'accessLevels'");
            }
            if ((!args || args.groupKey === undefined) && !opts.urn) {
                throw new Error("Missing required property 'groupKey'");
            }
            if ((!args || args.organizationId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'organizationId'");
            }
            resourceInputs["accessLevels"] = args ? args.accessLevels : undefined;
            resourceInputs["groupKey"] = args ? args.groupKey : undefined;
            resourceInputs["organizationId"] = args ? args.organizationId : undefined;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GcpUserAccessBinding.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GcpUserAccessBinding = GcpUserAccessBinding;
/** @internal */
GcpUserAccessBinding.__pulumiType = 'gcp:accesscontextmanager/gcpUserAccessBinding:GcpUserAccessBinding';
