// @pulumi/gcp/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.endpoints = exports.dns = exports.diagflow = exports.deploymentmanager = exports.datastream = exports.datastore = exports.dataproc = exports.dataplex = exports.dataloss = exports.datafusion = exports.dataform = exports.dataflow = exports.datacatalog = exports.databasemigrationservice = exports.containeranalysis = exports.container = exports.config = exports.compute = exports.composer = exports.cloudtasks = exports.cloudscheduler = exports.cloudrunv2 = exports.cloudrun = exports.cloudids = exports.cloudidentity = exports.cloudfunctionsv2 = exports.cloudfunctions = exports.clouddeploy = exports.cloudbuildv2 = exports.cloudbuild = exports.cloudasset = exports.certificatemanager = exports.certificateauthority = exports.binaryauthorization = exports.billing = exports.bigtable = exports.bigquerydatapolicy = exports.bigqueryanalyticshub = exports.bigquery = exports.beyondcorp = exports.assuredworkloads = exports.artifactregistry = exports.appengine = exports.apigee = exports.apigateway = exports.alloydb = exports.activedirectory = exports.accesscontextmanager = exports.accessapproval = exports.Provider = void 0;
exports.vertex = exports.types = exports.tpu = exports.tags = exports.storage = exports.sql = exports.spanner = exports.sourcerepo = exports.serviceusage = exports.servicenetworking = exports.servicedirectory = exports.serviceaccount = exports.securitycenter = exports.secretmanager = exports.runtimeconfig = exports.resourcemanager = exports.redis = exports.recaptcha = exports.pubsub = exports.projects = exports.oslogin = exports.osconfig = exports.orgpolicy = exports.organizations = exports.notebooks = exports.networkservices = exports.networksecurity = exports.networkmanagement = exports.networkconnectivity = exports.monitoring = exports.ml = exports.memcache = exports.logging = exports.kms = exports.iot = exports.identityplatform = exports.iap = exports.iam = exports.healthcare = exports.gkeonprem = exports.gkehub = exports.gkebackup = exports.gameservices = exports.folder = exports.firestore = exports.firebaserules = exports.firebase = exports.filestore = exports.eventarc = exports.essentialcontacts = void 0;
exports.workstations = exports.workflows = exports.vpcaccess = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
exports.Provider = null;
utilities.lazyLoad(exports, ["Provider"], () => require("./provider"));
__exportStar(require("./utils"), exports);
// Export sub-modules:
const accessapproval = require("./accessapproval");
exports.accessapproval = accessapproval;
const accesscontextmanager = require("./accesscontextmanager");
exports.accesscontextmanager = accesscontextmanager;
const activedirectory = require("./activedirectory");
exports.activedirectory = activedirectory;
const alloydb = require("./alloydb");
exports.alloydb = alloydb;
const apigateway = require("./apigateway");
exports.apigateway = apigateway;
const apigee = require("./apigee");
exports.apigee = apigee;
const appengine = require("./appengine");
exports.appengine = appengine;
const artifactregistry = require("./artifactregistry");
exports.artifactregistry = artifactregistry;
const assuredworkloads = require("./assuredworkloads");
exports.assuredworkloads = assuredworkloads;
const beyondcorp = require("./beyondcorp");
exports.beyondcorp = beyondcorp;
const bigquery = require("./bigquery");
exports.bigquery = bigquery;
const bigqueryanalyticshub = require("./bigqueryanalyticshub");
exports.bigqueryanalyticshub = bigqueryanalyticshub;
const bigquerydatapolicy = require("./bigquerydatapolicy");
exports.bigquerydatapolicy = bigquerydatapolicy;
const bigtable = require("./bigtable");
exports.bigtable = bigtable;
const billing = require("./billing");
exports.billing = billing;
const binaryauthorization = require("./binaryauthorization");
exports.binaryauthorization = binaryauthorization;
const certificateauthority = require("./certificateauthority");
exports.certificateauthority = certificateauthority;
const certificatemanager = require("./certificatemanager");
exports.certificatemanager = certificatemanager;
const cloudasset = require("./cloudasset");
exports.cloudasset = cloudasset;
const cloudbuild = require("./cloudbuild");
exports.cloudbuild = cloudbuild;
const cloudbuildv2 = require("./cloudbuildv2");
exports.cloudbuildv2 = cloudbuildv2;
const clouddeploy = require("./clouddeploy");
exports.clouddeploy = clouddeploy;
const cloudfunctions = require("./cloudfunctions");
exports.cloudfunctions = cloudfunctions;
const cloudfunctionsv2 = require("./cloudfunctionsv2");
exports.cloudfunctionsv2 = cloudfunctionsv2;
const cloudidentity = require("./cloudidentity");
exports.cloudidentity = cloudidentity;
const cloudids = require("./cloudids");
exports.cloudids = cloudids;
const cloudrun = require("./cloudrun");
exports.cloudrun = cloudrun;
const cloudrunv2 = require("./cloudrunv2");
exports.cloudrunv2 = cloudrunv2;
const cloudscheduler = require("./cloudscheduler");
exports.cloudscheduler = cloudscheduler;
const cloudtasks = require("./cloudtasks");
exports.cloudtasks = cloudtasks;
const composer = require("./composer");
exports.composer = composer;
const compute = require("./compute");
exports.compute = compute;
const config = require("./config");
exports.config = config;
const container = require("./container");
exports.container = container;
const containeranalysis = require("./containeranalysis");
exports.containeranalysis = containeranalysis;
const databasemigrationservice = require("./databasemigrationservice");
exports.databasemigrationservice = databasemigrationservice;
const datacatalog = require("./datacatalog");
exports.datacatalog = datacatalog;
const dataflow = require("./dataflow");
exports.dataflow = dataflow;
const dataform = require("./dataform");
exports.dataform = dataform;
const datafusion = require("./datafusion");
exports.datafusion = datafusion;
const dataloss = require("./dataloss");
exports.dataloss = dataloss;
const dataplex = require("./dataplex");
exports.dataplex = dataplex;
const dataproc = require("./dataproc");
exports.dataproc = dataproc;
const datastore = require("./datastore");
exports.datastore = datastore;
const datastream = require("./datastream");
exports.datastream = datastream;
const deploymentmanager = require("./deploymentmanager");
exports.deploymentmanager = deploymentmanager;
const diagflow = require("./diagflow");
exports.diagflow = diagflow;
const dns = require("./dns");
exports.dns = dns;
const endpoints = require("./endpoints");
exports.endpoints = endpoints;
const essentialcontacts = require("./essentialcontacts");
exports.essentialcontacts = essentialcontacts;
const eventarc = require("./eventarc");
exports.eventarc = eventarc;
const filestore = require("./filestore");
exports.filestore = filestore;
const firebase = require("./firebase");
exports.firebase = firebase;
const firebaserules = require("./firebaserules");
exports.firebaserules = firebaserules;
const firestore = require("./firestore");
exports.firestore = firestore;
const folder = require("./folder");
exports.folder = folder;
const gameservices = require("./gameservices");
exports.gameservices = gameservices;
const gkebackup = require("./gkebackup");
exports.gkebackup = gkebackup;
const gkehub = require("./gkehub");
exports.gkehub = gkehub;
const gkeonprem = require("./gkeonprem");
exports.gkeonprem = gkeonprem;
const healthcare = require("./healthcare");
exports.healthcare = healthcare;
const iam = require("./iam");
exports.iam = iam;
const iap = require("./iap");
exports.iap = iap;
const identityplatform = require("./identityplatform");
exports.identityplatform = identityplatform;
const iot = require("./iot");
exports.iot = iot;
const kms = require("./kms");
exports.kms = kms;
const logging = require("./logging");
exports.logging = logging;
const memcache = require("./memcache");
exports.memcache = memcache;
const ml = require("./ml");
exports.ml = ml;
const monitoring = require("./monitoring");
exports.monitoring = monitoring;
const networkconnectivity = require("./networkconnectivity");
exports.networkconnectivity = networkconnectivity;
const networkmanagement = require("./networkmanagement");
exports.networkmanagement = networkmanagement;
const networksecurity = require("./networksecurity");
exports.networksecurity = networksecurity;
const networkservices = require("./networkservices");
exports.networkservices = networkservices;
const notebooks = require("./notebooks");
exports.notebooks = notebooks;
const organizations = require("./organizations");
exports.organizations = organizations;
const orgpolicy = require("./orgpolicy");
exports.orgpolicy = orgpolicy;
const osconfig = require("./osconfig");
exports.osconfig = osconfig;
const oslogin = require("./oslogin");
exports.oslogin = oslogin;
const projects = require("./projects");
exports.projects = projects;
const pubsub = require("./pubsub");
exports.pubsub = pubsub;
const recaptcha = require("./recaptcha");
exports.recaptcha = recaptcha;
const redis = require("./redis");
exports.redis = redis;
const resourcemanager = require("./resourcemanager");
exports.resourcemanager = resourcemanager;
const runtimeconfig = require("./runtimeconfig");
exports.runtimeconfig = runtimeconfig;
const secretmanager = require("./secretmanager");
exports.secretmanager = secretmanager;
const securitycenter = require("./securitycenter");
exports.securitycenter = securitycenter;
const serviceaccount = require("./serviceaccount");
exports.serviceaccount = serviceaccount;
const servicedirectory = require("./servicedirectory");
exports.servicedirectory = servicedirectory;
const servicenetworking = require("./servicenetworking");
exports.servicenetworking = servicenetworking;
const serviceusage = require("./serviceusage");
exports.serviceusage = serviceusage;
const sourcerepo = require("./sourcerepo");
exports.sourcerepo = sourcerepo;
const spanner = require("./spanner");
exports.spanner = spanner;
const sql = require("./sql");
exports.sql = sql;
const storage = require("./storage");
exports.storage = storage;
const tags = require("./tags");
exports.tags = tags;
const tpu = require("./tpu");
exports.tpu = tpu;
const types = require("./types");
exports.types = types;
const vertex = require("./vertex");
exports.vertex = vertex;
const vpcaccess = require("./vpcaccess");
exports.vpcaccess = vpcaccess;
const workflows = require("./workflows");
exports.workflows = workflows;
const workstations = require("./workstations");
exports.workstations = workstations;
pulumi.runtime.registerResourcePackage("gcp", {
    version: utilities.getVersion(),
    constructProvider: (name, type, urn) => {
        if (type !== "pulumi:providers:gcp") {
            throw new Error(`unknown provider type ${type}`);
        }
        return new exports.Provider(name, undefined, { urn });
    },
});
