// @pulumi/gcp/sourcerepo/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RepositoryIamPolicy = exports.RepositoryIamMember = exports.RepositoryIamBinding = exports.Repository = exports.getRepositoryOutput = exports.getRepository = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getRepository = null;
exports.getRepositoryOutput = null;
utilities.lazyLoad(exports, ["getRepository", "getRepositoryOutput"], () => require("./getRepository"));
exports.Repository = null;
utilities.lazyLoad(exports, ["Repository"], () => require("./repository"));
exports.RepositoryIamBinding = null;
utilities.lazyLoad(exports, ["RepositoryIamBinding"], () => require("./repositoryIamBinding"));
exports.RepositoryIamMember = null;
utilities.lazyLoad(exports, ["RepositoryIamMember"], () => require("./repositoryIamMember"));
exports.RepositoryIamPolicy = null;
utilities.lazyLoad(exports, ["RepositoryIamPolicy"], () => require("./repositoryIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:sourcerepo/repository:Repository":
                return new exports.Repository(name, undefined, { urn });
            case "gcp:sourcerepo/repositoryIamBinding:RepositoryIamBinding":
                return new exports.RepositoryIamBinding(name, undefined, { urn });
            case "gcp:sourcerepo/repositoryIamMember:RepositoryIamMember":
                return new exports.RepositoryIamMember(name, undefined, { urn });
            case "gcp:sourcerepo/repositoryIamPolicy:RepositoryIamPolicy":
                return new exports.RepositoryIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "sourcerepo/repository", _module);
pulumi.runtime.registerResourceModule("gcp", "sourcerepo/repositoryIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "sourcerepo/repositoryIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "sourcerepo/repositoryIamPolicy", _module);
