// @pulumi/gcp/tags/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagValueIamPolicy = exports.TagValueIamMember = exports.TagValueIamBinding = exports.TagValue = exports.TagKeyIamPolicy = exports.TagKeyIamMember = exports.TagKeyIamBinding = exports.TagKey = exports.TagBinding = exports.LocationTagBinding = exports.getTagValueOutput = exports.getTagValue = exports.getTagKeyOutput = exports.getTagKey = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.getTagKey = null;
exports.getTagKeyOutput = null;
utilities.lazyLoad(exports, ["getTagKey", "getTagKeyOutput"], () => require("./getTagKey"));
exports.getTagValue = null;
exports.getTagValueOutput = null;
utilities.lazyLoad(exports, ["getTagValue", "getTagValueOutput"], () => require("./getTagValue"));
exports.LocationTagBinding = null;
utilities.lazyLoad(exports, ["LocationTagBinding"], () => require("./locationTagBinding"));
exports.TagBinding = null;
utilities.lazyLoad(exports, ["TagBinding"], () => require("./tagBinding"));
exports.TagKey = null;
utilities.lazyLoad(exports, ["TagKey"], () => require("./tagKey"));
exports.TagKeyIamBinding = null;
utilities.lazyLoad(exports, ["TagKeyIamBinding"], () => require("./tagKeyIamBinding"));
exports.TagKeyIamMember = null;
utilities.lazyLoad(exports, ["TagKeyIamMember"], () => require("./tagKeyIamMember"));
exports.TagKeyIamPolicy = null;
utilities.lazyLoad(exports, ["TagKeyIamPolicy"], () => require("./tagKeyIamPolicy"));
exports.TagValue = null;
utilities.lazyLoad(exports, ["TagValue"], () => require("./tagValue"));
exports.TagValueIamBinding = null;
utilities.lazyLoad(exports, ["TagValueIamBinding"], () => require("./tagValueIamBinding"));
exports.TagValueIamMember = null;
utilities.lazyLoad(exports, ["TagValueIamMember"], () => require("./tagValueIamMember"));
exports.TagValueIamPolicy = null;
utilities.lazyLoad(exports, ["TagValueIamPolicy"], () => require("./tagValueIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:tags/locationTagBinding:LocationTagBinding":
                return new exports.LocationTagBinding(name, undefined, { urn });
            case "gcp:tags/tagBinding:TagBinding":
                return new exports.TagBinding(name, undefined, { urn });
            case "gcp:tags/tagKey:TagKey":
                return new exports.TagKey(name, undefined, { urn });
            case "gcp:tags/tagKeyIamBinding:TagKeyIamBinding":
                return new exports.TagKeyIamBinding(name, undefined, { urn });
            case "gcp:tags/tagKeyIamMember:TagKeyIamMember":
                return new exports.TagKeyIamMember(name, undefined, { urn });
            case "gcp:tags/tagKeyIamPolicy:TagKeyIamPolicy":
                return new exports.TagKeyIamPolicy(name, undefined, { urn });
            case "gcp:tags/tagValue:TagValue":
                return new exports.TagValue(name, undefined, { urn });
            case "gcp:tags/tagValueIamBinding:TagValueIamBinding":
                return new exports.TagValueIamBinding(name, undefined, { urn });
            case "gcp:tags/tagValueIamMember:TagValueIamMember":
                return new exports.TagValueIamMember(name, undefined, { urn });
            case "gcp:tags/tagValueIamPolicy:TagValueIamPolicy":
                return new exports.TagValueIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "tags/locationTagBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagKey", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagKeyIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagKeyIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagKeyIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagValue", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagValueIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagValueIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "tags/tagValueIamPolicy", _module);
