// @pulumi/gcp/apigateway/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GatewayIamPolicy = exports.GatewayIamMember = exports.GatewayIamBinding = exports.Gateway = exports.ApiIamPolicy = exports.ApiIamMember = exports.ApiIamBinding = exports.ApiConfigIamPolicy = exports.ApiConfigIamMember = exports.ApiConfigIamBinding = exports.ApiConfig = exports.Api = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Api = null;
utilities.lazyLoad(exports, ["Api"], () => require("./api"));
exports.ApiConfig = null;
utilities.lazyLoad(exports, ["ApiConfig"], () => require("./apiConfig"));
exports.ApiConfigIamBinding = null;
utilities.lazyLoad(exports, ["ApiConfigIamBinding"], () => require("./apiConfigIamBinding"));
exports.ApiConfigIamMember = null;
utilities.lazyLoad(exports, ["ApiConfigIamMember"], () => require("./apiConfigIamMember"));
exports.ApiConfigIamPolicy = null;
utilities.lazyLoad(exports, ["ApiConfigIamPolicy"], () => require("./apiConfigIamPolicy"));
exports.ApiIamBinding = null;
utilities.lazyLoad(exports, ["ApiIamBinding"], () => require("./apiIamBinding"));
exports.ApiIamMember = null;
utilities.lazyLoad(exports, ["ApiIamMember"], () => require("./apiIamMember"));
exports.ApiIamPolicy = null;
utilities.lazyLoad(exports, ["ApiIamPolicy"], () => require("./apiIamPolicy"));
exports.Gateway = null;
utilities.lazyLoad(exports, ["Gateway"], () => require("./gateway"));
exports.GatewayIamBinding = null;
utilities.lazyLoad(exports, ["GatewayIamBinding"], () => require("./gatewayIamBinding"));
exports.GatewayIamMember = null;
utilities.lazyLoad(exports, ["GatewayIamMember"], () => require("./gatewayIamMember"));
exports.GatewayIamPolicy = null;
utilities.lazyLoad(exports, ["GatewayIamPolicy"], () => require("./gatewayIamPolicy"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:apigateway/api:Api":
                return new exports.Api(name, undefined, { urn });
            case "gcp:apigateway/apiConfig:ApiConfig":
                return new exports.ApiConfig(name, undefined, { urn });
            case "gcp:apigateway/apiConfigIamBinding:ApiConfigIamBinding":
                return new exports.ApiConfigIamBinding(name, undefined, { urn });
            case "gcp:apigateway/apiConfigIamMember:ApiConfigIamMember":
                return new exports.ApiConfigIamMember(name, undefined, { urn });
            case "gcp:apigateway/apiConfigIamPolicy:ApiConfigIamPolicy":
                return new exports.ApiConfigIamPolicy(name, undefined, { urn });
            case "gcp:apigateway/apiIamBinding:ApiIamBinding":
                return new exports.ApiIamBinding(name, undefined, { urn });
            case "gcp:apigateway/apiIamMember:ApiIamMember":
                return new exports.ApiIamMember(name, undefined, { urn });
            case "gcp:apigateway/apiIamPolicy:ApiIamPolicy":
                return new exports.ApiIamPolicy(name, undefined, { urn });
            case "gcp:apigateway/gateway:Gateway":
                return new exports.Gateway(name, undefined, { urn });
            case "gcp:apigateway/gatewayIamBinding:GatewayIamBinding":
                return new exports.GatewayIamBinding(name, undefined, { urn });
            case "gcp:apigateway/gatewayIamMember:GatewayIamMember":
                return new exports.GatewayIamMember(name, undefined, { urn });
            case "gcp:apigateway/gatewayIamPolicy:GatewayIamPolicy":
                return new exports.GatewayIamPolicy(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "apigateway/api", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiConfigIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiConfigIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiConfigIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/apiIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/gateway", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/gatewayIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/gatewayIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "apigateway/gatewayIamPolicy", _module);
