// @pulumi/gcp/bigquery/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Table = exports.Routine = exports.ReservationAssignment = exports.Reservation = exports.Job = exports.IamPolicy = exports.IamMember = exports.IamBinding = exports.getDefaultServiceAccountOutput = exports.getDefaultServiceAccount = exports.DatasetIamPolicy = exports.DatasetIamMember = exports.DatasetIamBinding = exports.DatasetAccess = exports.Dataset = exports.DataTransferConfig = exports.ConnectionIamPolicy = exports.ConnectionIamMember = exports.ConnectionIamBinding = exports.Connection = exports.CapacityCommitment = exports.AppProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.AppProfile = null;
utilities.lazyLoad(exports, ["AppProfile"], () => require("./appProfile"));
exports.CapacityCommitment = null;
utilities.lazyLoad(exports, ["CapacityCommitment"], () => require("./capacityCommitment"));
exports.Connection = null;
utilities.lazyLoad(exports, ["Connection"], () => require("./connection"));
exports.ConnectionIamBinding = null;
utilities.lazyLoad(exports, ["ConnectionIamBinding"], () => require("./connectionIamBinding"));
exports.ConnectionIamMember = null;
utilities.lazyLoad(exports, ["ConnectionIamMember"], () => require("./connectionIamMember"));
exports.ConnectionIamPolicy = null;
utilities.lazyLoad(exports, ["ConnectionIamPolicy"], () => require("./connectionIamPolicy"));
exports.DataTransferConfig = null;
utilities.lazyLoad(exports, ["DataTransferConfig"], () => require("./dataTransferConfig"));
exports.Dataset = null;
utilities.lazyLoad(exports, ["Dataset"], () => require("./dataset"));
exports.DatasetAccess = null;
utilities.lazyLoad(exports, ["DatasetAccess"], () => require("./datasetAccess"));
exports.DatasetIamBinding = null;
utilities.lazyLoad(exports, ["DatasetIamBinding"], () => require("./datasetIamBinding"));
exports.DatasetIamMember = null;
utilities.lazyLoad(exports, ["DatasetIamMember"], () => require("./datasetIamMember"));
exports.DatasetIamPolicy = null;
utilities.lazyLoad(exports, ["DatasetIamPolicy"], () => require("./datasetIamPolicy"));
exports.getDefaultServiceAccount = null;
exports.getDefaultServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getDefaultServiceAccount", "getDefaultServiceAccountOutput"], () => require("./getDefaultServiceAccount"));
exports.IamBinding = null;
utilities.lazyLoad(exports, ["IamBinding"], () => require("./iamBinding"));
exports.IamMember = null;
utilities.lazyLoad(exports, ["IamMember"], () => require("./iamMember"));
exports.IamPolicy = null;
utilities.lazyLoad(exports, ["IamPolicy"], () => require("./iamPolicy"));
exports.Job = null;
utilities.lazyLoad(exports, ["Job"], () => require("./job"));
exports.Reservation = null;
utilities.lazyLoad(exports, ["Reservation"], () => require("./reservation"));
exports.ReservationAssignment = null;
utilities.lazyLoad(exports, ["ReservationAssignment"], () => require("./reservationAssignment"));
exports.Routine = null;
utilities.lazyLoad(exports, ["Routine"], () => require("./routine"));
exports.Table = null;
utilities.lazyLoad(exports, ["Table"], () => require("./table"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:bigquery/appProfile:AppProfile":
                return new exports.AppProfile(name, undefined, { urn });
            case "gcp:bigquery/capacityCommitment:CapacityCommitment":
                return new exports.CapacityCommitment(name, undefined, { urn });
            case "gcp:bigquery/connection:Connection":
                return new exports.Connection(name, undefined, { urn });
            case "gcp:bigquery/connectionIamBinding:ConnectionIamBinding":
                return new exports.ConnectionIamBinding(name, undefined, { urn });
            case "gcp:bigquery/connectionIamMember:ConnectionIamMember":
                return new exports.ConnectionIamMember(name, undefined, { urn });
            case "gcp:bigquery/connectionIamPolicy:ConnectionIamPolicy":
                return new exports.ConnectionIamPolicy(name, undefined, { urn });
            case "gcp:bigquery/dataTransferConfig:DataTransferConfig":
                return new exports.DataTransferConfig(name, undefined, { urn });
            case "gcp:bigquery/dataset:Dataset":
                return new exports.Dataset(name, undefined, { urn });
            case "gcp:bigquery/datasetAccess:DatasetAccess":
                return new exports.DatasetAccess(name, undefined, { urn });
            case "gcp:bigquery/datasetIamBinding:DatasetIamBinding":
                return new exports.DatasetIamBinding(name, undefined, { urn });
            case "gcp:bigquery/datasetIamMember:DatasetIamMember":
                return new exports.DatasetIamMember(name, undefined, { urn });
            case "gcp:bigquery/datasetIamPolicy:DatasetIamPolicy":
                return new exports.DatasetIamPolicy(name, undefined, { urn });
            case "gcp:bigquery/iamBinding:IamBinding":
                return new exports.IamBinding(name, undefined, { urn });
            case "gcp:bigquery/iamMember:IamMember":
                return new exports.IamMember(name, undefined, { urn });
            case "gcp:bigquery/iamPolicy:IamPolicy":
                return new exports.IamPolicy(name, undefined, { urn });
            case "gcp:bigquery/job:Job":
                return new exports.Job(name, undefined, { urn });
            case "gcp:bigquery/reservation:Reservation":
                return new exports.Reservation(name, undefined, { urn });
            case "gcp:bigquery/reservationAssignment:ReservationAssignment":
                return new exports.ReservationAssignment(name, undefined, { urn });
            case "gcp:bigquery/routine:Routine":
                return new exports.Routine(name, undefined, { urn });
            case "gcp:bigquery/table:Table":
                return new exports.Table(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "bigquery/appProfile", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/capacityCommitment", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/connection", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/connectionIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/connectionIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/connectionIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/dataTransferConfig", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/dataset", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/datasetAccess", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/datasetIamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/datasetIamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/datasetIamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/iamBinding", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/iamMember", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/iamPolicy", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/job", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/reservation", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/reservationAssignment", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/routine", _module);
pulumi.runtime.registerResourceModule("gcp", "bigquery/table", _module);
