// @pulumi/gcp/appengine/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StandardAppVersion = exports.ServiceNetworkSettings = exports.getDefaultServiceAccountOutput = exports.getDefaultServiceAccount = exports.FlexibleAppVersion = exports.FirewallRule = exports.EngineSplitTraffic = exports.DomainMapping = exports.ApplicationUrlDispatchRules = exports.Application = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.Application = null;
utilities.lazyLoad(exports, ["Application"], () => require("./application"));
exports.ApplicationUrlDispatchRules = null;
utilities.lazyLoad(exports, ["ApplicationUrlDispatchRules"], () => require("./applicationUrlDispatchRules"));
exports.DomainMapping = null;
utilities.lazyLoad(exports, ["DomainMapping"], () => require("./domainMapping"));
exports.EngineSplitTraffic = null;
utilities.lazyLoad(exports, ["EngineSplitTraffic"], () => require("./engineSplitTraffic"));
exports.FirewallRule = null;
utilities.lazyLoad(exports, ["FirewallRule"], () => require("./firewallRule"));
exports.FlexibleAppVersion = null;
utilities.lazyLoad(exports, ["FlexibleAppVersion"], () => require("./flexibleAppVersion"));
exports.getDefaultServiceAccount = null;
exports.getDefaultServiceAccountOutput = null;
utilities.lazyLoad(exports, ["getDefaultServiceAccount", "getDefaultServiceAccountOutput"], () => require("./getDefaultServiceAccount"));
exports.ServiceNetworkSettings = null;
utilities.lazyLoad(exports, ["ServiceNetworkSettings"], () => require("./serviceNetworkSettings"));
exports.StandardAppVersion = null;
utilities.lazyLoad(exports, ["StandardAppVersion"], () => require("./standardAppVersion"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:appengine/application:Application":
                return new exports.Application(name, undefined, { urn });
            case "gcp:appengine/applicationUrlDispatchRules:ApplicationUrlDispatchRules":
                return new exports.ApplicationUrlDispatchRules(name, undefined, { urn });
            case "gcp:appengine/domainMapping:DomainMapping":
                return new exports.DomainMapping(name, undefined, { urn });
            case "gcp:appengine/engineSplitTraffic:EngineSplitTraffic":
                return new exports.EngineSplitTraffic(name, undefined, { urn });
            case "gcp:appengine/firewallRule:FirewallRule":
                return new exports.FirewallRule(name, undefined, { urn });
            case "gcp:appengine/flexibleAppVersion:FlexibleAppVersion":
                return new exports.FlexibleAppVersion(name, undefined, { urn });
            case "gcp:appengine/serviceNetworkSettings:ServiceNetworkSettings":
                return new exports.ServiceNetworkSettings(name, undefined, { urn });
            case "gcp:appengine/standardAppVersion:StandardAppVersion":
                return new exports.StandardAppVersion(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "appengine/application", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/applicationUrlDispatchRules", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/domainMapping", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/engineSplitTraffic", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/firewallRule", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/flexibleAppVersion", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/serviceNetworkSettings", _module);
pulumi.runtime.registerResourceModule("gcp", "appengine/standardAppVersion", _module);
