// @pulumi/gcp/dataloss/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PreventionStoredInfoType = exports.PreventionJobTrigger = exports.PreventionInspectTemplate = exports.PreventionDeidentifyTemplate = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.PreventionDeidentifyTemplate = null;
utilities.lazyLoad(exports, ["PreventionDeidentifyTemplate"], () => require("./preventionDeidentifyTemplate"));
exports.PreventionInspectTemplate = null;
utilities.lazyLoad(exports, ["PreventionInspectTemplate"], () => require("./preventionInspectTemplate"));
exports.PreventionJobTrigger = null;
utilities.lazyLoad(exports, ["PreventionJobTrigger"], () => require("./preventionJobTrigger"));
exports.PreventionStoredInfoType = null;
utilities.lazyLoad(exports, ["PreventionStoredInfoType"], () => require("./preventionStoredInfoType"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "gcp:dataloss/preventionDeidentifyTemplate:PreventionDeidentifyTemplate":
                return new exports.PreventionDeidentifyTemplate(name, undefined, { urn });
            case "gcp:dataloss/preventionInspectTemplate:PreventionInspectTemplate":
                return new exports.PreventionInspectTemplate(name, undefined, { urn });
            case "gcp:dataloss/preventionJobTrigger:PreventionJobTrigger":
                return new exports.PreventionJobTrigger(name, undefined, { urn });
            case "gcp:dataloss/preventionStoredInfoType:PreventionStoredInfoType":
                return new exports.PreventionStoredInfoType(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionDeidentifyTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionInspectTemplate", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionJobTrigger", _module);
pulumi.runtime.registerResourceModule("gcp", "dataloss/preventionStoredInfoType", _module);
