// @pulumi/gcp/kms/registry.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Registry = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
/**
 * A Google Cloud IoT Core device registry.
 *
 * To get more information about DeviceRegistry, see:
 *
 * * [API documentation](https://cloud.google.com/iot/docs/reference/cloudiot/rest/)
 * * How-to Guides
 *     * [Official Documentation](https://cloud.google.com/iot/docs/)
 *
 * ## Example Usage
 * ### Cloudiot Device Registry Basic
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const test_registry = new gcp.iot.Registry("test-registry", {});
 * ```
 * ### Cloudiot Device Registry Single Event Notification Configs
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as gcp from "@pulumi/gcp";
 *
 * const default_telemetry = new gcp.pubsub.Topic("default-telemetry", {});
 * const test_registry = new gcp.iot.Registry("test-registry", {eventNotificationConfigs: [{
 *     pubsubTopicName: default_telemetry.id,
 *     subfolderMatches: "",
 * }]});
 * ```
 * ### Cloudiot Device Registry Full
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as fs from "fs";
 * import * as gcp from "@pulumi/gcp";
 *
 * const default_devicestatus = new gcp.pubsub.Topic("default-devicestatus", {});
 * const default_telemetry = new gcp.pubsub.Topic("default-telemetry", {});
 * const additional_telemetry = new gcp.pubsub.Topic("additional-telemetry", {});
 * const test_registry = new gcp.iot.Registry("test-registry", {
 *     eventNotificationConfigs: [
 *         {
 *             pubsubTopicName: additional_telemetry.id,
 *             subfolderMatches: "test/path",
 *         },
 *         {
 *             pubsubTopicName: default_telemetry.id,
 *             subfolderMatches: "",
 *         },
 *     ],
 *     stateNotificationConfig: {
 *         pubsub_topic_name: default_devicestatus.id,
 *     },
 *     mqttConfig: {
 *         mqtt_enabled_state: "MQTT_ENABLED",
 *     },
 *     httpConfig: {
 *         http_enabled_state: "HTTP_ENABLED",
 *     },
 *     logLevel: "INFO",
 *     credentials: [{
 *         publicKeyCertificate: {
 *             format: "X509_CERTIFICATE_PEM",
 *             certificate: fs.readFileSync("test-fixtures/rsa_cert.pem"),
 *         },
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * DeviceRegistry can be imported using any of these accepted formats
 *
 * ```sh
 *  $ pulumi import gcp:kms/registry:Registry default {{project}}/locations/{{region}}/registries/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:kms/registry:Registry default {{project}}/{{region}}/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:kms/registry:Registry default {{region}}/{{name}}
 * ```
 *
 * ```sh
 *  $ pulumi import gcp:kms/registry:Registry default {{name}}
 * ```
 *
 * @deprecated gcp.kms.Registry has been deprecated in favor of gcp.iot.Registry
 */
class Registry extends pulumi.CustomResource {
    /**
     * Get an existing Registry resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("Registry is deprecated: gcp.kms.Registry has been deprecated in favor of gcp.iot.Registry");
        return new Registry(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of Registry.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Registry.__pulumiType;
    }
    /** @deprecated gcp.kms.Registry has been deprecated in favor of gcp.iot.Registry */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("Registry is deprecated: gcp.kms.Registry has been deprecated in favor of gcp.iot.Registry");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["credentials"] = state ? state.credentials : undefined;
            resourceInputs["eventNotificationConfigs"] = state ? state.eventNotificationConfigs : undefined;
            resourceInputs["httpConfig"] = state ? state.httpConfig : undefined;
            resourceInputs["logLevel"] = state ? state.logLevel : undefined;
            resourceInputs["mqttConfig"] = state ? state.mqttConfig : undefined;
            resourceInputs["name"] = state ? state.name : undefined;
            resourceInputs["project"] = state ? state.project : undefined;
            resourceInputs["region"] = state ? state.region : undefined;
            resourceInputs["stateNotificationConfig"] = state ? state.stateNotificationConfig : undefined;
        }
        else {
            const args = argsOrState;
            resourceInputs["credentials"] = args ? args.credentials : undefined;
            resourceInputs["eventNotificationConfigs"] = args ? args.eventNotificationConfigs : undefined;
            resourceInputs["httpConfig"] = args ? args.httpConfig : undefined;
            resourceInputs["logLevel"] = args ? args.logLevel : undefined;
            resourceInputs["mqttConfig"] = args ? args.mqttConfig : undefined;
            resourceInputs["name"] = args ? args.name : undefined;
            resourceInputs["project"] = args ? args.project : undefined;
            resourceInputs["region"] = args ? args.region : undefined;
            resourceInputs["stateNotificationConfig"] = args ? args.stateNotificationConfig : undefined;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Registry.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Registry = Registry;
/** @internal */
Registry.__pulumiType = 'gcp:kms/registry:Registry';
